src/planner/builder/builder.go

```diff
@@ -91,22 +86,18 @@ func processSelect(log *xlog.Log, router *router.Router, database string, node *
 	}
 
 	if node.Having != nil {
-		havings, err := parseHaving(node.Having.Expr, tbInfos, root.getFields())
-		if err != nil {
-			return nil, err
-		}
-		if err = root.pushHaving(havings); err != nil {
+		if err = pushHavings(root, node.Having.Expr, tbInfos); err != nil {
 			return nil, err
 		}
 	}
 
-	if err = root.pushOrderBy(node); err != nil {
+	if err = root.pushOrderBy(node.OrderBy); err != nil {
 		return nil, err
 	}
 
 	// Limit SubPlan.
 	if node.Limit != nil {
-		if err = root.pushLimit(node); err != nil {
+		if err = root.pushLimit(node.Limit); err != nil {
```

Comment:
Move `if node.Limit != nil` to the `pushLimit`?

Reply:
If the limit is nil, why we called the function. 
if move to `pushlimit`, `JoinNode` `MergeNode` `UnionNode`, `SubNode`, all of them need  add this judgment at the beginning, which is not worth to do it.

Reply:
I saw the `if len(node.OrderBy) > 0` in the `pushOrderBy`. 
you need keep them consistent.

Reply:
pushOrderBy has been modified. Force update yesterday.